src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs

```diff
@@ -160,6 +207,18 @@ public static Diagnostic Filter(Diagnostic d, int warningLevelOption, ReportDiag
             }
 
             return report;
+
+            bool promoteToAnError()
+            {
+                Debug.Assert(report == ReportDiagnostic.Default);
```

Comment:
>Debug [](start = 16, length = 5)

Should this method also assert `generalDiagnosticOption == ReportDiagnostic.Error`?